[Docs] Note the order of enumerated custom keycodes (#15706)
qmk/qmk_firmware: docs/feature_macros.md

};
```

> ?> Enumerated list of custom keycodes (`enum custom_keycodes`) must always be declared before `keymaps[]` array, `process_record_user()` and any other function that uses them for the compiler to recognise the list.

Reviewer: I feel like this implies that the enum must be added multiple times before each use case. Maybe something like this would be better? Certainly open to other options.
```suggestion
?> The enumerated list of custom keycodes `enum custom_keycodes` must be declared only once and before the `keymaps[]` array or any other function that uses custom keycodes.
```

Author: Updated sentence grammar to reference a single list to avoid ambiguity.
